### PR TITLE
[REFACTOR] Drop Python 2 support and update pylint

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
Drop Python 2 as per https://github.com/dmlc/gluon-nlp/issues/721.

This allows updating pylint of which our current version emits false positive E1111 with https://github.com/dmlc/gluon-nlp/pull/836.

Also fixes https://github.com/dmlc/gluon-nlp/issues/839.

### Essentials ###
- [X] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented

### Changes ###
- [X] Drop Python 2 support and update pylint

### CODE REVIEW

**Comment by mli** [comment]
Found link check problems in job PR-838/1:
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-838/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-838/1/index.html

**Comment by mli** [comment]
Found link check problems in job PR-838/3:
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-838/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-838/3/index.html

**Comment by mli** [comment]
Found link check problems in job PR-838/4:
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-838/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-838/4/index.html

File: src/gluonnlp/embedding/evaluation.py [hunk 1]
```diff
@@ -32,12 +32,10 @@
 
 class _WordEmbeddingEvaluationFunction(HybridBlock):  # pylint: disable=abstract-method
     """Base class for word embedding evaluation functions."""
-    pass
```
Comment (sxjscience) [inline src/gluonnlp/embedding/evaluation.py]: Why should we remove the `pass` here?

Comment (leezu) [reply src/gluonnlp/embedding/evaluation.py]: It's not required. Lint fails (pylint 2 introduced a new rule)

**Pull request merged**